### PR TITLE
Cythonize

### LINKED ISSUE #9: Cythonized build?
The current mistune package is pure Python, and doesn't rely on Cython for performance improvements.  I noticed this first in the downstream [Anaconda package](https://github.com/ContinuumIO/anaconda-issues/issues/9215), where older builds (>=0.7.3) did rely on Cython and emit per-platform builds. Is the desire here to retain this as a noarch package, or would adding Cython be acceptable for mistune? It adds [a bit of performance](https://github.com/lepture/mistune/issues/1) to the package, and is simple enough to implement. I'll submit a PR if this is a desired change.

### PR DESCRIPTION
**Pull request by scw**
Convert the package to a arch-based one, and use Cython for improved performance. This resolves #9.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,12 +9,12 @@ source:
   sha256: bc10c33bfdcaa4e749b779f62f60d6e12f8215c46a292d05e486b869ae306619
 
 build:
-  number: 0
-  noarch: python
-  script: python -m pip install --no-deps --ignore-installed .
+  number: 1
+  script: python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (ocefpaf) [inline recipe/meta.yaml]: I believe you don't need to move it back to the old way. `pip` should build the extension correctly here.
Can you revert this change?

Comment (scw) [reply recipe/meta.yaml]: OK, switched back to the original form.

**Comment by ocefpaf** [comment]
Thanks @scw!

**Comment by bollwyvl** [comment]
Awesome!
